common/src/main/java/com/wynntils/overlays/StatusEffectsOverlay.java

```diff
@@ -108,4 +134,51 @@ private void updateTextRenderSetting() {
     protected TextRenderSetting getTextRenderSetting() {
         return textRenderSetting;
     }
+
+    private Stream<RenderedStatusEffect> stackEffects(List<StatusEffect> effects) {
+        Map<String, RenderedStatusEffect> effectsToRender = new LinkedHashMap<>();
+
+        for (StatusEffect effect : effects) {
+            RenderedStatusEffect entry =
+                    effectsToRender.get(effect.asString().getString());
+            if (entry == null) {
+                entry = new RenderedStatusEffect(effect);
+                effectsToRender.put(effect.asString().getString(), entry);
+            }
+            entry.count += 1;
+        }
+        return effectsToRender.values().stream();
+    }
+
+    private static final class RenderedStatusEffect {
+        private int count = 0;
+        private final StatusEffect effect;
+
+        private RenderedStatusEffect(StyledText name, StyledText modifier, StyledText displayedTime, StyledText prefix) {
+            this.effect = new StatusEffect(name, modifier, displayedTime, prefix);
+        }
+
+        private RenderedStatusEffect(StatusEffect effect) {
+            this.effect = effect;
+        }
+
+        private StyledText getTextToRender() {
+            return this.count > 1
+                    ? StyledText.fromString(this.count + "x ").append(this.effect.asString())
+                    : this.effect.asString();
+        }
+
+        public int getCount() {
+            return this.count;
+        }
+
+        public void setCount(int c) {
+            this.count = c;
+        }
+
```

Comment:
You added this but you still use
```
            entry.count += 1;
```

Reply:
Ah, forgot about that one. I thought it would give me errors where it changed but I guess it has access to the inner parts of the class.